### ISSUE TITLE
User can see 502 Bad Gateway in case of an unstable connection

### ISSUE THREAD

**Issue by Roald87** [Roald87/motherfucking-rss-reader#15]
This error was logged while I tried to access the site over an unstable connection. It was probably due to the fact that I lost internet connection while the request was ongoing. A request can take some time to process since it takes a while before all the feeds are gathered.

I don't think there is a way to get around this without implementing Javascript or other fancy solutions, that go against the philosophy of this project.

```
 ---> System.Net.HttpListenerException (0x80131620): Unable to write data to the transport connection: Broken pipe.
   at System.Net.HttpResponseStream.EndWriteCore(IAsyncResult asyncResult)
   at System.IO.Stream.<>c.<BeginEndWriteAsync>b__62_1(Stream stream, IAsyncResult asyncResult)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncTrimPromise`1.Complete(TInstance thisRef, Func`3 endMethod, IAsyncResult asyncResult, Boolean requiresSynchronization)
   --- End of inner exception stack trace ---
   at Microsoft.FSharp.Control.AsyncResult`1.Commit() in D:\a\_work\1\s\src\FSharp.Core\async.fs:line 454
   at Microsoft.FSharp.Control.AsyncPrimitives.QueueAsyncAndWaitForResultSynchronously[a](CancellationToken token, FSharpAsync`1 computation, FSharpOption`1 timeout) in D:\a\_work\1\s\src\FSharp.Core\async.fs:line 1139
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronously[T](CancellationToken cancellationToken, FSharpAsync`1 computation, FSharpOption`1 timeout) in D:\a\_work\1\s\src\FSharp.Core\async.fs:line 1166
   at Microsoft.FSharp.Control.FSharpAsync.RunSynchronously[T](FSharpAsync`1 computation, FSharpOption`1 timeout, FSharpOption`1 cancellationToken) in D:\a\_work\1\s\src\FSharp.Core\async.fs:line 1515
   at Program.main(String[] argv) in /root/motherfucking-rss-reader/SimpleRssServer/Program.fs:line 39
/bin/bash: line 1: /usr/bin/ts: No such file or directory

```

**Comment by Roald87** [comment]
See https://stackoverflow.com/a/26942784/6329629

**Comment by Roald87** [comment]
A possible solution would be to regularly fetch most requested feeds. Then they are always up-to-date and the response time is almost always short.

**Comment by Roald87** [comment]
Closed with #23
